low-level/files/opt/seagate/sspl/setup/sspl_prepare.py

```diff
@@ -0,0 +1,127 @@
+                "storage_enclosure>%s>storage>controller>primary>ip" % enclosure_id)
+            secondary_ip = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "storage_enclosure>%s>storage>controller>secondary>ip" % enclosure_id)
+            cntrlr_user = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "storage_enclosure>%s>storage>controller>user" % enclosure_id)
+            cntrlr_passwd = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "storage_enclosure>%s>storage>controller>password" % enclosure_id)
+            data_private_fqdn = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "server_node>%s>network>data>private_fqdn" % machine_id)
+            data_private_interfaces = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "server_node>%s>network>data>private_interfaces" % machine_id)
+            data_public_fqdn = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "server_node>%s>network>data>public_fqdn" % machine_id)
+            data_public_interfaces = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "server_node>%s>network>data>public_interfaces" % machine_id)
+            # mgmt public fqdn should also be for VM?
+            mgmt_public_fqdn = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+                "server_node>%s>network>management>public_fqdn" % machine_id)
+        mgmt_interfaces = Utility.get_config_value(PRVSNR_CONFIG_INDEX,
+            "server_node>%s>network>management>interfaces" % machine_id)
+
+        # Validate BMC & Storage controller accessibility
+        if node_type.lower() != "vm":
+            BmcV().validate('accessible', [socket.getfqdn(), bmc_ip, bmc_user, bmc_passwd])
+            c_validator = ControllerV()
+            c_validator.validate("accessible", [primary_ip, cntrlr_user, cntrlr_passwd])
+            c_validator.validate("accessible", [secondary_ip, cntrlr_user, cntrlr_passwd])
+
+            # Validate network fqdn reachability
+            # mgmt public fqdn should also be for VM?
+            NetworkV().validate("connectivity", [mgmt_public_fqdn,
+                data_private_fqdn, data_public_fqdn])
+
+        # Validate network interface availability
+        for i_list in [mgmt_interfaces, data_private_interfaces, data_public_interfaces]:
+            self.validate_nw_interface(i_list)
+
+    def process(self):
+        """Configure SSPL at prepare stage."""
+        pass
+
+    def validate_nw_interface(self, interfaces):
+        """Check network interfaces are up.
+        carrier:0 - link down
+        carrier:1 - link up
+        """
+        if not isinstance(interfaces, list):
+            raise SetupError(errno.EINVAL, "%s - validation failure. %s",
+                self.name, "Expected list of interfaces. Received, %s." % interfaces)
+        for interface in interfaces:
+            cmd = "cat /sys/class/net/%s/carrier" % interface
```

Comment:
I think 'carrier' was for cable. Additionally can use 'ip' command & check interface reachability

Reply:
Okay

Reply:
Done.